### PR TITLE
[xDS] move CDS and EDS watchers into xds resolver

### PR DESCRIPTION
**Pull request by markdroth**
Implements gRFC A74 (https://github.com/grpc/proposal/pull/404).

### CODE REVIEW

**Comment by markdroth** [comment]
@yashykt, please review the changes related to the xDS mTLS support.

@eugeneo, please review the PR overall.

Thanks!

**Review by eugeneo — APPROVED** [review]
lgtm

File: src/core/ext/filters/client_channel/resolver/xds/xds_config.h [hunk 1]
```diff
@@ -0,0 +1,247 @@
+//
```
[the remark this answers is not in the document]
Comment (markdroth) [reply src/core/ext/filters/client_channel/resolver/xds/xds_config.h]: Done.

File: src/core/ext/filters/client_channel/resolver/xds/xds_resolver_trace.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+//
+// Copyright 2019 gRPC authors.
```
[the remark this answers is not in the document]
Comment (markdroth) [reply src/core/ext/filters/client_channel/resolver/xds/xds_resolver_trace.h:2]: When moving code from an existing file to a new file, I think we should retain the original copyright date.

**Review by yashykt — APPROVED** [review]

**Comment by yashykt** [comment]
Can I request that we do some 1000x runs just to be sure that we are not introducing any flakiness?

**Comment by markdroth** [comment]
I ran the following tests from test/cpp/end2end/xds on RBE 1000x in a debug build for both the epoll1 and poll pollers:
- xds_end2end_test
- xds_core_end2end_test
- xds_cluster_end2end_test
- xds_cluster_type_end2end_test
- xds_outlier_detection_end2end_test
- xds_override_host_end2end_test
- xds_rls_end2end_test

There were only a couple of flakes, and both of them are pre-existing.

I'll run this again overnight tonight under MSAN to see if anything else shakes out.

**Comment by markdroth** [comment]
The MSAN run finally finished (took more than 24 hrs), and the only flake was a pre-existing one, so I don't think there's much risk of introducing flakes here.

I will not plan to merge this until after the holidays.

**Comment by markdroth** [comment]
@eugeneo, please take another look, since I made some changes since you approved based on discussions in the gRFC PR.  Thanks!

**Review by eugeneo — APPROVED** [review]

**Pull request closed without merging**